src/Backend/Core/Engine/Base/Config.php

```diff
@@ -127,36 +137,77 @@ public function setModule($module)
     }
 
     /**
-     * Set the possible actions, based on files in folder
-     * You can disable action in the config file. (Populate $disabledActions)
+     * @return array
+     */
+    public function getPossibleActionTypes()
+    {
+        return array(
+            'actions',
+            'ajax',
+        );
+    }
+
+    /**
+     * @param string $action
+     * @return bool
      */
-    public function setPossibleActions()
+    public function isActionAvailable($action)
     {
-        $path = BACKEND_MODULES_PATH . '/' . $this->getModule();
-        if (is_dir($path . '/Actions')) {
-            $finder = new Finder();
-            foreach ($finder->files()->name('*.php')->in($path . '/Actions') as $file) {
-                /** @var $file \SplFileInfo */
-                $action = str_replace('.php', '', $file->getBasename());
-
-                // if the action isn't disabled add it to the possible actions
-                if (!in_array($action, $this->disabledActions)) {
-                    $this->possibleActions[$file->getBasename()] = $action;
-                }
+        // Save our action
+        $this->action = $action;
+
+        // Loop over every action type
+        foreach ($this->getPossibleActionTypes() as $actionType) {
+            // If this action is disabled for this type, continue on to the next type
+            if ($this->isActionDisabled($actionType)) {
+                continue;
             }
+
+            // If the action file is missing, continue on to the next type
+            if (!$this->isActionFilePresent($actionType)) {
+                continue;
+            }
+
+            // The action is not disabled and the file is preset, this is an available action!
+            return true;
         }
 
-        if (is_dir($path . '/Ajax')) {
```

Comment:
It's funny this code contained checks for Ajax actions, since this is never executed when doing ajax requests (as far as I can see on first sight)